components/camel-kafka/src/main/java/org/apache/camel/component/kafka/KafkaConsumer.java

```diff
@@ -423,6 +427,12 @@ public void onPartitionsAssigned(Collection<TopicPartition> partitions) {
         }
     }
 
+    private void propagateHeaders(ConsumerRecord<Object, Object> record, Exchange exchange, HeaderFilterStrategy headerFilterStrategy) {
+        StreamSupport.stream(record.headers().spliterator(), false)
+                .filter(header -> !headerFilterStrategy.applyFilterToCamelHeaders(header.key(), header.value(), exchange))
```

Comment:
filter(keepIncludedHeaders)?

Reply:
ok, will add additional method for this purpose